hal/src/stm32f2xx/usb_hal.c

```diff
@@ -83,7 +83,7 @@ static USBD_Class_cb_TypeDef* USB_Composite_Instance = NULL;
 
 #ifdef USB_VENDOR_REQUEST_ENABLE
 
-#define USB_VENDOR_REQUEST_TIMEOUT 1000
+#define USB_VENDOR_REQUEST_TIMEOUT 30000
```

Comment:
30 seconds feels too long for a request to get cancelled

Reply:
It's probably fine. Worst that can happen is that it'll be hanging around for those 30 seconds, which isn't a big deal. At least with CDC I've seen the host (especially on Windows) not processing transfers for over a couple of seconds.